File: frontend/src/api/tools.ts
/**
 * Tools API client
 */

const API_BASE = '/api';

export interface ToolInfo {
  name: string;
  description: string;
  parameters: Record<string, unknown>;
  source_file: string;
}

export interface ToolsConfig {
  tools: ToolInfo[];
}

export type ToolSelections = Record<string, boolean>;

/**
 * Get all available tools
 */
export async function getTools(): Promise<ToolsConfig> {
  const response = await fetch(`${API_BASE}/tools`);
  if (!response.ok) {
    throw new Error('Failed to get tools');
  }
  return response.json();
}

/**
 * Get a specific tool by name
 */
export async function getTool(name: string): Promise<ToolInfo> {
  const response = await fetch(`${API_BASE}/tools/${encodeURIComponent(name)}`);
  if (!response.ok) {
    throw new Error('Failed to get tool');
  }
  return response.json();
}

/**
 * Force refresh tools from disk
 */
export async function refreshTools(): Promise<ToolsConfig> {
  const response = await fetch(`${API_BASE}/tools/refresh`, { method: 'POST' });
  if (!response.ok) {
    throw new Error('Failed to refresh tools');
  }
  return response.json();
}

File: src/copilot_console/app/routers/tools.py
"""Local Tools API router."""

from fastapi import APIRouter, HTTPException

from copilot_console.app.models.tools import ToolInfo, ToolsConfig
from copilot_console.app.services.tools_service import get_tools_service

router = APIRouter(prefix="/tools", tags=["tools"])


@router.get("", response_model=ToolsConfig)
async def get_tools() -> ToolsConfig:
    """Get all available tools."""
    service = get_tools_service()
    return service.get_tools_config()


@router.get("/{name}", response_model=ToolInfo)
async def get_tool(name: str) -> ToolInfo:
    """Get a specific tool by name."""
    service = get_tools_service()
    tool = service.get_tool(name)
    
    if not tool:
        raise HTTPException(status_code=404, detail=f"Tool '{name}' not found")
    
    return tool


@router.post("/refresh", response_model=ToolsConfig)
async def refresh_tools() -> ToolsConfig:
    """Force refresh tools from disk."""
    service = get_tools_service()
    return service.refresh()
